sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java

```diff
@@ -251,4 +251,23 @@ default Context getSharedSpanBuilder(String spanName, Context context) {
      */
     default void addEvent(String name, Map<String, Object> attributes, OffsetDateTime timestamp) {
```

Comment:
Are we not deprecating this as @trask  suggested? I agree that this interface design is centered around passing the `Context` to each method. So, this method should be deprecated and still use the default behavior to use the current span if someone calls it.